src/consensus.rs

```diff
@@ -373,16 +314,8 @@ impl Consensus {
     /// and have enough stake
     /// and have a valid signature
     fn verify_new_bonded_validators(&mut self, proposal: &ConsensusProposal) -> Result<()> {
-        let proposal_pubkeys = proposal
-            .new_bonded_validators
-            .clone()
-            .into_iter()
-            .map(|c| c.pubkey)
-            .collect::<Vec<ValidatorPublicKey>>();
-        if proposal.slot != 0 && proposal_pubkeys != proposal.block.new_bonded_validators {
-            bail!("New bonded validators in proposal and block do not match. Proposal: {:?}, block: {:?}",
-                proposal_pubkeys, 
-                proposal.block.new_bonded_validators);
+        if proposal.slot != 0 {
+            return Ok(());
```

Comment:
please remove this condition :) with this condition, the function is never executed

Reply:
removed.